droidlet/interpreter/robot/default_behaviors.py

```diff
@@ -36,18 +37,29 @@ def init_logger():
 
 init_logger()
 
+#TODO: Move these utils to a suitable place
+def add_or_replace(agent, pred_text, obj_text):
+    memids, _ = agent.memory.basic_search(f'SELECT uuid FROM Triple WHERE pred_text={pred_text}')
+    assert len(memids) <= 1, f"more than 1 {len(memids)} returned"
+    if len(memids) > 0:
+        agent.memory.forget(memids[0])
+    agent.memory.add_triple(subj=agent.memory.self_memid, pred_text=pred_text, obj_text=obj_text)
+
+def get_unique_val_from_memory(agent, pred_text, typ):
```

Comment:
I don't understand why you need this?

Reply:
this isn't new functionality, just moved the memory lookup into a function. Other than the lookup, this does type casting as well and asserts for failures.